src/gui/dialogs/liveadjust_z.cpp

```diff
@@ -22,11 +23,87 @@ const float z_offset_step = 1.0F / float(axis_steps_per_unit[2]);
 const float z_offset_min = Z_OFFSET_MIN;
 const float z_offset_max = Z_OFFSET_MAX;
 
+/*****************************************************************************/
+//WindowScale
+
+WindowScale::WindowScale(window_t *parent, point_i16_t pt)
+    : AddSuperWindow<window_frame_t>(parent, Rect16(pt, 10, 100))
+    , scaleNum0(parent, getNumRect(pt), 0)
+    , scaleNum1(parent, getNumRect(pt), -1)
+    , scaleNum2(parent, getNumRect(pt), -2) {
+
+    scaleNum0.SetFont(GuiDefaults::Font);
+    scaleNum1.SetFont(GuiDefaults::Font);
+    scaleNum2.SetFont(GuiDefaults::Font);
+    scaleNum0.SetFormat("% .0f");
+
+    scaleNum0.rect -= Rect16::Top_t(8);
+    scaleNum1.rect += Rect16::Top_t((rect.Height() / 2) - 8);
+    scaleNum2.rect += Rect16::Top_t(rect.Height() - 8);
+
+    point = pt;
+    movePercent = 0.0F;
+}
+
+const Rect16 WindowScale::getNumRect(point_i16_t pt) {
+    return Rect16(pt.x - 30, pt.y, 30, 20);
+}
+
+void WindowScale::SetPercent(float p) {
+    uint16_t old_y = point.y + (rect.Height() * movePercent);
+    uint16_t new_y = point.y + (rect.Height() * p);
+    movePercent = p;
+    if (old_y != new_y) {
+        // instead of fill rectangle and draw scale again
+        // we just draw black line over orange one's old position
+        display::DrawLine(
```

Comment:
do not Draw inside `SetPercent `method, no one can expect this, 
`SetPercent `should just store old value, set new one and call `Invalidate()`